### PR TITLE
Enable the use of Yarn for package management

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR introduces the ability to use Yarn for package management.

In the root Unlock directory, simply run `yarn` and it will install deps for all apps. (might be worth doing `yarn nuke` first).

On a first install, it takes about 8 minutes to download and install everything. Once the cache is full, subsequent installs take less than 2 minutes.

We'll need to talk about using yarn with our CI and docker images and whatnot in the future, right now this is only present for local dev.

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [x] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,24 @@
   "version": "1.0.0",
   "description": "Unlock is a protocol for decentralized access control",
   "main": "index.js",
+  "private": true,
```
Comment (cnasc) [inline package.json]: Repo has to be marked private to enable use of workspaces.

File: package.json [hunk 2]
```diff
@@ -3,6 +3,24 @@
...
+    "rover",
+    "smart-contracts",
+    "tests",
+    "tickets",
+    "unlock-app",
+    "unlock-js",
+    "unlock-protocol.com",
+    "wedlocks"
+  ],
+  "resolutions": {
```
Comment (cnasc) [inline package.json]: See: https://yarnpkg.com/lang/en/docs/selective-version-resolutions/

Basically, this makes everything use the same version of React no matter which it actually specifies.

File: unlock-protocol.com/package.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "unlock-app",
+  "name": "unlock-protocol.com",
```
Comment (cnasc) [inline unlock-protocol.com/package.json]: Duplicate name made yarn angry

File: package.json (hunk 1 above)
Comment (julien51) [reply package.json]: Does it have any other consequnce?

File: unlock-protocol.com/package.json (hunk 3 above)
Comment (julien51) [reply unlock-protocol.com/package.json]: 😠

File: package.json (hunk 1 above)
Comment (cnasc) [reply package.json]: I _believe_ it would only impact publishing this parent module on npm but I will double check.

Comment (cnasc) [reply package.json]: Confirmed, this would only prevent publishing the root-level unlock module. Does not interfere with subrepos.

**Comment by cnasc** [comment]
Just did another push -- it looks like snapshots are different between the yarn and npm environments (just css class names). If that is the case and my recent push didn't fix things, we'll need to update CI to use yarn to move this forward.

**Comment by cnasc** [comment]
This is pretty much dead in the water, things run fine locally but then resolution of the storybook modules fail in the docker image. Not worth burning more time on.

**Comment by julien51** [comment]
snapshots!

**Comment by cnasc** [comment]
> snapshots!

IIRC we have the issue here where snapshots generated with Yarn will be different from the ones done with NPM. Should be able to fix it so that we just get deps with Yarn for now and run the scripts with NPM still

**Comment by cnasc** [comment]
Closing in favor of #5613

**Pull request closed without merging**